ant-networking/src/bootstrap.rs

```diff
@@ -18,12 +18,57 @@ use libp2p::{
 use rand::seq::SliceRandom;
 use std::collections::{HashSet, VecDeque};
 
+/// Periodically check if the initial bootstrap process should be triggered.
+/// This happens only once after the conditions for triggering the initial bootstrap process are met.
+pub(crate) const INITIAL_BOOTSTRAP_CHECK_INTERVAL: std::time::Duration =
+    std::time::Duration::from_secs(1);
+
 /// The max number of concurrent dials to be made during the initial bootstrap process.
 const CONCURRENT_DIALS: usize = 3;
 
 /// The max number of peers to be added to the routing table before stopping the initial bootstrap process.
 const MAX_PEERS_BEFORE_TERMINATION: usize = 5;
 
+/// This is used to track the conditions that are required to trigger the initial bootstrap process once.
+pub(crate) struct InitialBootstrapTrigger {
+    pub(crate) upnp: bool,
+    pub(crate) client: bool,
+    pub(crate) upnp_gateway_result_obtained: bool,
+    pub(crate) listen_addr_obtained: bool,
+}
```

Comment:
Chose to not merge this with the other struct as I get pretty confused when I try to do that..